alchemiscale/interface/api.py

```diff
@@ -128,43 +147,140 @@ def query_networks(
+
+    return [str(sk) for sk in results]
+
+
+@router.get("/networks/{network_scoped_key}/transformations")
+def get_network_transformations(
     network_scoped_key,
     *,
     n4js: Neo4jStore = Depends(get_n4js_depends),
     token: TokenData = Depends(get_token_data_depends),
 ):
-    # Get scope from scoped key provided by user, uniquely identifying the network
     sk = ScopedKey.from_str(network_scoped_key)
     validate_scopes(sk.scope, token)
 
-    network = n4js.get_gufe(scoped_key=sk)
-    return gufe_to_json(network)
+    return [str(sk) for sk in n4js.get_network_transformations(network=sk)]
 
 
-@router.post("/networks", response_model=ScopedKey)
-def create_network(
+@router.get("/transformations/{transformation_scoped_key}/networks")
+def get_transformation_networks(
+    transformation_scoped_key,
     *,
-    network: Dict = Body(...),
-    scope: Scope,
     n4js: Neo4jStore = Depends(get_n4js_depends),
     token: TokenData = Depends(get_token_data_depends),
 ):
-    validate_scopes(scope, token)
+    sk = ScopedKey.from_str(transformation_scoped_key)
+    validate_scopes(sk.scope, token)
 
-    an = AlchemicalNetwork.from_dict(network)
-    an_sk = n4js.create_network(network=an, scope=scope)
+    return [str(sk) for sk in n4js.get_transformation_networks(transformation=sk)]
 
-    # create taskhub for this network
-    n4js.create_taskhub(an_sk)
 
-    return an_sk
+@router.get("/networks/{network_scoped_key}/chemicalsystems")
+def get_network_chemicalsystems(
+    network_scoped_key,
+    *,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    token: TokenData = Depends(get_token_data_depends),
+):
+    sk = ScopedKey.from_str(network_scoped_key)
+    validate_scopes(sk.scope, token)
 
+    return [str(sk) for sk in n4js.get_network_chemicalsystems(network=sk)]
+
+
+@router.get("/chemicalsystems/{chemicalsystem_scoped_key}/networks")
+def get_chemicalsystem_networks(
+    chemicalsystem_scoped_key,
+    *,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    token: TokenData = Depends(get_token_data_depends),
+):
+    sk = ScopedKey.from_str(chemicalsystem_scoped_key)
+    validate_scopes(sk.scope, token)
+
+    return [str(sk) for sk in n4js.get_chemicalsystem_networks(chemicalsystem=sk)]
 
-@router.get("/transformations")
-def query_transformations():
-    return {"message": "nothing yet"}
+
+@router.get("/transformations/{transformation_scoped_key}/chemicalsystems")
+def get_transformation_chemicalsystems(
+    transformation_scoped_key,
+    *,
+    n4js: Neo4jStore = Depends(get_n4js_depends),
+    token: TokenData = Depends(get_token_data_depends),
+):
+    sk = ScopedKey.from_str(transformation_scoped_key)
+    validate_scopes(sk.scope, token)
+
+    return [
+        str(sk) for sk in n4js.get_transformation_chemicalsystems(transformation=sk)
+    ]
+
+
+@router.get("/chemicalsystems/{chemicalsystem_scoped_key}/transformations")
+def get_chemicalsystem_transformations(
+    chemicalsystem_scoped_key,
```

Comment:
For the second case, would it give back a list of lists, corresponding to the order of `ScopedKey`s given in `chemsys_sks`? 

I suppose it could be a useful pattern to give back only a single item if a single item is given, an iterable of an iterable is given. For the iterable case, it may make sense for it to perform `async` calls for greater performance.

I'd rather expand these methods to do multi support if there is user demand. There are relatively few cases where I think there's high demand upfront (e.g. setting or getting many `Task` statuses), so we'll have to see how users are bottlenecked in practice. Agree?

Reply:
Sounds good to me :)